contrib: Add test for ELF symbol-check

laanwj opened the pull request:
Check both failure cases:
- Use a glibc symbol from a version that is too new
- Use a symbol from a library that is not in the allowlist

And also check a conforming binary.

Adding a similar check for Windows PE can be done in a separate PR.

laanwj committed:
contrib: Add test for ELF symbol-check

Check both failure cases:
- Use a glibc symbol from a version that is too new
- Use a symbol from a library that is not in the allowlist

And also check a conforming binary.

Adding a similar check for Windows PE can be done in a separate PR.

fanquake commented on contrib/devtools/test-symbol-check.py:
```diff
@@ -0,0 +1,88 @@
+        if 'riscv' in get_machine(cc):
+            self.skipTest("test not available for RISC-V")
+
+        # emfd_create was introduced in GLIBC 2.27, so is newer than the upper limit of
```
s/emfd_create/memfd_create

fanquake reviewed (commented):
Concept ACK - thanks for putting this together. Will actually test shortly.

fanquake committed:
contrib: add MACHO tests to symbol-check tests

fanquake commented:
I added a test for macOS if you wanted to pull that in as well: https://github.com/fanquake/bitcoin/commit/9e6e59b01e1e4e3325bab59cf97139b462b42651.

laanwj commented:
> I added a test for macOS if you wanted to pull that in as well: fanquake@9e6e59b.

Thanks ! I didn't mention macOS in the OP because I (mistakenly) thought we only had a symbol check for windows and linux. Good to have a test for that too.

I've fixed the comment and cherry-picked that commit.

laanwj commented:
(close/ropen was to test the IRC bot)

fanquake reviewed (approved):
ACK ed1bbcefeaafef460b6e5609de85e13c12e0d2a4
